### PR TITLE
[expo-update][ios] Fix manifest headers in multipart responses

### PR DESCRIPTION
**Pull request by wschurman**
# Why

iOS equivalent of https://github.com/expo/expo/pull/15497.

# How

Same strategy as https://github.com/expo/expo/pull/15497 (add manifest-response-format-agnostic container for headers).

# Test Plan

Same as https://github.com/expo/expo/pull/15497.

### CODE REVIEW

File: packages/expo-updates/ios/EXUpdates/AppLoader/EXUpdatesFileDownloader.m [hunk 1]
```diff
@@ -143,8 +144,14 @@ - (void)parseManifestResponse:(NSHTTPURLResponse *)httpResponse
                                    successBlock:successBlock
                                      errorBlock:errorBlock];
   } else {
+    NSDictionary *responseHeaders = [httpResponse allHeaderFields];
+    EXUpdatesManifestHeaders *manifestHeaders = [[EXUpdatesManifestHeaders alloc] initWithProtocolVersion:responseHeaders[@"expo-protocol-version"]
```
Comment (ide) [inline packages/expo-updates/ios/EXUpdates/AppLoader/EXUpdatesFileDownloader.m:148]: I feel better about using valueForHTTPHeaderField instead, which handles case-insensitivity compared to allHeaderFields trying to canonicalize casing (it doesn't just lowercase)

Comment (wschurman) [reply packages/expo-updates/ios/EXUpdates/AppLoader/EXUpdatesFileDownloader.m:148]: Unfortunately it's only available on iOS 13+: https://developer.apple.com/documentation/foundation/nshttpurlresponse/3240613-valueforhttpheaderfield?language=objc

Comment (ide) [reply packages/expo-updates/ios/EXUpdates/AppLoader/EXUpdatesFileDownloader.m:148]: We'll probably drop iOS 12 sometime after the iOS 16 release or perhaps earlier (Google Maps is dropping it) so I'd leave a comment here saying we can use the method after dropping iOS 12.

**Review by ide — APPROVED** [review]

File: packages/expo-updates/ios/EXUpdates/AppLoader/EXUpdatesFileDownloader.m [hunk 2]
```diff
@@ -143,8 +144,14 @@ - (void)parseManifestResponse:(NSHTTPURLResponse *)httpResponse
                                    successBlock:successBlock
                                      errorBlock:errorBlock];
   } else {
+    NSDictionary *responseHeaders = [httpResponse allHeaderFields];
```
Comment (wschurman) [inline packages/expo-updates/ios/EXUpdates/AppLoader/EXUpdatesFileDownloader.m:147]:
```suggestion
    // can use valueForHTTPHeaderField after dropping iOS 12 support
    NSDictionary *responseHeaders = [httpResponse allHeaderFields];
```

**Pull request merged**